gnoblog-cli/cmd/post.go

```diff
@@ -0,0 +1,261 @@
+	if cfg.Quiet {
+		pass, err = io.GetPassword("", cfg.InsecurePasswordStdIn)
+	} else {
+		pass, err = io.GetPassword("Enter password:", cfg.InsecurePasswordStdIn)
+	}
+	if err != nil {
+		return err
+	}
+
+	// Initialize Gnoclient
+	client := gnoclient.Client{
+		Signer:    initSigner(cfg, pass),
+		RPCClient: initRPCClient(cfg),
+	}
+
+	// Batch post request passed in
+	if fileInfo.IsDir() {
+		return cfg.batchPost(client, args[0])
+	}
+
+	// Single post request passed in an argument
+	return cfg.singlePost(client, args[0])
+}
+
+func (cfg *cliCfg) singlePost(c gnoclient.Client, postPath string) error {
+	postFile, err := os.Open(postPath)
+	if err != nil {
+		return fmt.Errorf("cannot open file %q: %w", postPath, err)
+	}
+
+	post, err := parsePost(postFile)
+	if err != nil {
+		return fmt.Errorf("cannot parse post %q: %w", postPath, err)
+	}
+
+	signingAcc, _, err := c.QueryAccount(c.Signer.Info().GetAddress())
+	if err != nil {
+		return err
+	}
+
+	nonce := signingAcc.GetSequence()
+	accNumber := signingAcc.GetAccountNumber()
+
+	// Check if post already exists
+	existsExpr := "PostExists(\"" + post.Slug + "\")"
+	exists, _, err := c.QEval(cfg.BlogRealmPath, existsExpr)
+
+	verb := "ModAddPost"
+	if strings.Contains(exists, "true") && cfg.Edit {
+		verb = "ModEditPost"
+	}
+
+	msg := gnoclient.MsgCall{
+		PkgPath:  cfg.BlogRealmPath,
+		FuncName: verb,
+		Args: []string{
+			post.Slug,
+			post.Title,
+			post.Body,
+			post.PublicationDate.Format(time.RFC3339),
+			strings.Join(post.Authors, ","),
+			strings.Join(post.Tags, ","),
+		},
+	}
+
+	baseTxCfg := gnoclient.BaseTxCfg{
+		GasFee:         cfg.GasFee,
+		GasWanted:      cfg.GasWanted,
+		AccountNumber:  accNumber,
+		SequenceNumber: nonce,
+		Memo:           "Posted from gnoblog-cli",
+	}
+
+	if !cfg.Publish {
+		fmt.Println("--publish flag not set to true, exiting")
+		// todo see how to display generated data
+		return nil
+	}
+
+	fmt.Println(msg)
+	_, err = c.Call(baseTxCfg, msg)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Successfully posted %s", post.Title)
+
+	return nil
+}
+
+func (cfg *cliCfg) batchPost(c gnoclient.Client, dirPath string) error {
+	files, err := findFilePaths(dirPath)
+	if err != nil {
+		return err
+	}
+
+	msgs := make([]gnoclient.MsgCall, 0, len(files))
+	for _, postPath := range files {
+		postFile, err := os.Open(postPath)
+		if err != nil {
+			return fmt.Errorf("cannot open file %q: %w", postPath, err)
+		}
+
+		post, err := parsePost(postFile)
+		if err != nil {
+			return fmt.Errorf("cannot parse post %q: %w", postPath, err)
+		}
+
+		callCfg := gnoclient.MsgCall{
+			PkgPath:  cfg.BlogRealmPath,
+			FuncName: "ModAddPost",
+			Args: []string{
+				post.Slug,
+				post.Title,
+				post.Body,
+				post.PublicationDate.Format(time.RFC3339),
+				strings.Join(post.Authors, ","),
+				strings.Join(post.Tags, ","),
+			},
+		}
+		msgs = append(msgs, callCfg)
+	}
+
+	signingAcc, _, err := c.QueryAccount(c.Signer.Info().GetAddress())
```

Comment:
err not checked.

Reply:
You added the check here: [2f41a71](https://github.com/gnolang/blog/pull/63/commits/2f41a7196bcedc71e29208e0147a6ee36b09e1d8)